### PR TITLE
Migrate to BaseConnectorTest in BigQuery

### LINKED ISSUE #7110: Migrate BigQuery tests to use BaseConnectorTest
https://github.com/trinodb/trino/pull/6989 introduced new `BaseConnectorTest` test class which replaces `AbstractTestDistributedQueries` and `AbstractTestIntegrationSmokeTest`.

Scope of this ticket is to restructure BigQuery test code to follow new pattern.
`TestBigQueryIntegrationSmokeTest` should be renamed to `TestBigQueryConnectorTest` and extend `BaseConnectorTest` instead of `AbstractTestIntegrationSmokeTest`

A PR which restructures MySQL tests is a good reference:  https://github.com/trinodb/trino/pull/7011

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Migrate to BaseConnectorTest in BigQuery

* Fixes #7110

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java [hunk 1]
```diff
@@ -771,7 +771,7 @@ public void testDeleteWithVarcharPredicate()
     protected void skipTestUnlessSupportsDeletes()
     {
         skipTestUnless(supportsCreateTable());
-        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_supports_delete", "(col varchar(1))", ImmutableList.of("'a'", "'A'"))) {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_supports_delete", "(col varchar(1))")) {
```
Comment (hashhar) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java:774]: Thank you. I was testing for CREATE TABLE but performing CTAS in the test.

File: testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java [hunk 2]
```diff
@@ -1048,6 +1048,15 @@ public void testQueryLoggingCount()
         });
     }
 
+    /**
+     * Two columns (foo_1 and foo_2_4) must be used in the table definition
```
Comment (hashhar) [inline testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java:1052]: if any data type is fine then maybe reword like:

`The table must have two columns foo_1 and foo_2_4 of any type.`

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryConnectorTest.java [hunk 3]
```diff
@@ -192,6 +257,80 @@ public void testDropTable()
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
     }
 
+    @Test
+    @Override
+    public void testRenameTable()
+    {
+        // Use CREATE TABLE instead of CREATE TABLE AS statement
```
Comment (hashhar) [inline plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryConnectorTest.java:264]: Can we fix this in base test? It seems we don't need CTAS to test rename table either?

Comment (ebyhr) [reply plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryConnectorTest.java:264]: We could do it, but I hesitated to change in this PR because the test contains assertion for values.

**Review by hashhar — APPROVED** [review]
